google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITTransactionTest.java

```diff
@@ -349,4 +353,110 @@ public Void run(TransactionContext transaction) throws SpannerException {
                 .getLong(0))
         .isEqualTo(2);
   }
+
+  private void doNestedRwTransaction() {
+    client
+        .readWriteTransaction()
+        .run(
+            new TransactionCallable<Void>() {
+              @Override
+              public Void run(TransactionContext transaction) throws SpannerException {
+                client
+                    .readWriteTransaction()
+                    .run(
+                        new TransactionCallable<Void>() {
+                          @Override
+                          public Void run(TransactionContext transaction) throws Exception {
+                            return null;
+                          }
+                        });
+
+                return null;
+              }
+            });
+  }
+
+  @Test
+  public void nestedRwRwTransactionShouldThrowException() {
```

Comment:
RwRw a typo ?

also have RW (i assume its for ReadWrite)

Reply:
nit: I would prefer u shorten "TransactionShouldThrowException" over RW RD etc since those explain the test case..

Reply:
RwRw -> Nested Read-Write transaction within a Read-Write transaction.

Didn't quite understand the shorten comment. If I shorted the name, then all the 4 methods will have the same name?

Reply:
had offline conversation